command/server/config_test_helpers.go

```diff
@@ -861,6 +861,71 @@ listener "tcp" {
 	}
 }
 
+func testParseUserLockouts(t *testing.T) {
+	obj, _ := hcl.Parse(strings.TrimSpace(`
+	user_lockout "all" {
+		lockout_duration = "40m"
+		lockout_counter_reset = "45m"
+		disable_lockout = "false"
+	}
+	  user_lockout "userpass" {
+	     lockout_threshold = "100"
+	     lockout_duration = "20m"
+	  }
+	  user_lockout "ldap" {
+		disable_lockout = "true"
+	 }`))
+
+	config := Config{
+		SharedConfig: &configutil.SharedConfig{},
+	}
+	list, _ := obj.Node.(*ast.ObjectList)
+	objList := list.Filter("user_lockout")
+	configutil.ParseUserLockouts(config.SharedConfig, objList)
+
+	expected := &Config{
+		SharedConfig: &configutil.SharedConfig{
+			UserLockouts: []*configutil.UserLockout{
+				{
+					Type:                   "all",
+					LockoutThreshold:       5,
+					LockoutThresholdRaw:    nil,
```

Comment:
The test would failed if we remove this as it compares with what the values in SharedConfig would look like.

Reply:
I see. I am curious to know what is the failure. The zero value for an interface is nil, so if you don't set it nil explicitly, it should still be nil, no?

Reply:
Sorry, the test was flaky before and the comparison failed for config and expected config sometimes due to arrangement of array elements. When I removed raw entries then, the comparison failed and I felt it was because of that. I think the flaky test fix fixed that and on removing the nil fields, the comparison is successful.